lib/client/client.js

```diff
@@ -23,8 +23,8 @@ const run = (buildHash, options) => {
   const { replace } = require('./hmr');
   const { error, info, warn } = require('./log')();
 
-  const protocol = secure ? 'wss' : 'ws';
-  const socket = new ClientSocket(client, `${protocol}://${client.address || address}/wps`);
+  const serverProtocol = secure ? 'wss' : 'ws';
+  const socket = new ClientSocket(client, `${client.protocol || serverProtocol}://${client.address || address}/wps`);
```

Comment:
```suggestion
  const socket = new ClientSocket(client, `${client.protocol || protocol}://${client.address || address}/wps`);
```

I'm not sure what the benefit was to changing the name, but if it's not a scope issue (which it doesn't appear to be) it's best practice not to make "I like this better" changes.